corehq/ex-submodules/phonelog/utils.py

```diff
@@ -207,11 +210,11 @@ def _get_header(self, fmt):
         if settings.SERVER_ENVIRONMENT == 'production':
             environment = 'prod'
 
-        return {b"X-Sumo-Category": "{env}/{domain}/{fmt}".format(
-            env=environment,
-            domain=self.domain,
-            fmt=fmt,
-        ).encode('utf-8')}
+        header = "{env}/{domain}/{fmt}".format(env=environment, domain=self.domain, fmt=fmt)
+
+        if isTask:
+            return header
+        return {b"X-Sumo-Category": header.encode('utf-8')}
```

Comment:
Even though you've taken care of this by adding `b"X-Sumo-Category"` in the task itself, I am not the biggest fan of having to use `b"X-Sumo-Category` in multiple places.

This makes the `is_task` variable imply multiple things: whether or not data is encoded, _and_ whether or not a dict with the `X-Sumo-Category` or a string is returned from this method. I don't have a good alternative solution at the moment, so maybe that just shows that this is the best we can do. I think it is worth chatting with a more experienced dev just to see if there is a better alternative.

Reply:
Is there a master file or something similar that holds commonly used strings (encoded or not) that I could add this string to and import in instead?

Reply:
Ah my issue with this isn't that we are using the same string twice, although at some point putting a string in a const.py file for the app is a good idea if _needing_ to use it in multiple places.

However my issue here is that the `send_device_log_to_sumologic` task now needs to _know_ to include `X-Sumo-Category`. In other words, the `headers` value that was passed to this method needs more modification than just encoding/decoding before sending up. IMO, `_get_headers` should be the only place responsible for providing everything needed in the header.

Reply:
To keep __get_headers the only place where the dictionary is created, I've updated it so that it returns the whole dictionary as either encoded or not, and to encode both (the singular) key and value in the task.